Add 1-step continuous aggregate policies

charislam opened the pull request:
Update the documentation to include the new 1-step continuous aggregate
policies.

Adds:
*   An API reference for every policy
*   ~~A section in the compression, retention, and continuous aggregate
    refresh sections to reference the new features~~

Updates:
*   Updates troubleshooting frontmatter in the new directory to
    reference the new APIs where applicable

# Links

Fixes #1476 

# Writing help

For information about style and word usage, see the [style guide](https://docs.timescale.com/timescaledb/latest/contribute-to-docs/)

# Review checklists
Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

- [ ] Is the content technically accurate?
- [ ] Is the content complete?
- [ ] Is the content presented in a logical order?
- [ ] Does the content use appropriate names for features and products?
- [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

- [ ] Is the content free from typos?
- [ ] Does the content use plain English?
- [ ] Does the content contain clear sections for concepts, tasks, and references?
- [ ] Have any images been uploaded to the correct location, and are resolvable?
- [ ] If the page index was updated, are redirects required
      and have they been implemented?
- [ ] Have you checked the built version of this content?

charislam committed:
Add 1-step continuous aggregate policies

Update the documentation to include the new 1-step continuous aggregate
policies.

Adds:
*   An API reference for every policy
*   A section in the compression, retention, and continuous aggregate
    refresh sections to reference the new features

Updates:
*   Updates troubleeshooting frontmatter in the new directory to
    reference the new APIs where applicable

charislam commented on api/alter_policies.md:
```diff
@@ -0,0 +1,70 @@
+
+|Name|Type|Description|
+|-|-|-|
+|`if_not_exists`|`BOOL`|When true, prints a warning instead of erroring if the policy doesn't exist. Defaults to false.|
```
@RafiaSabih Is it the policy or the continuous aggregate that the `if_not_exists` applies to? I assumed the policy

RafiaSabih replied:
Yes, you are right, it is referring to policy.

charislam commented on api/policies.md:
```diff
@@ -0,0 +1,81 @@
+aggregates.
+
+<highlight type="note">
+Only policies applying to continuous aggregates are shown in this view. Policies
+applying to regular hypertables or regular materialized views are not displayed.
```
@RafiaSabih Is this true? This is what I got from reading the comments on the tests

RafiaSabih replied:
Yes.

charislam commented on api/remove_all_policies.md:
```diff
@@ -0,0 +1,54 @@
+
+|Name|Type|Description|
+|-|-|-|
+|`if_exists`|`BOOL`|When true, prints a warning instead of erroring if no policies exist. Defaults to false.|
```
@RafiaSabih Is it "if _no_ policies exist" or does it also error if any one of the policies doesn't exist?

charislam commented on api/remove_continuous_aggregate_policy.md:
```diff
@@ -9,23 +9,39 @@ api:
-Remove the refresh policy from the 'cpu_view' continuous aggregate:
+|Name|Type|Description|
+|-|-|-|
+|`if_not_exists`|`BOOL`|When true, prints a warning instead of erroring if the policy doesn't exist. Defaults to false. Deprecated in TimescaleDB 2.8.|
+|`if_exists`|`BOOL`|When true, prints a warning instead of erroring if the policy doesn't exist. Defaults to `NULL`, which is the same as false. Introduced in TimescaleDB 2.8.|
```
@RafiaSabih Not entirely sure I have this right

RafiaSabih replied:
This is just to change the name of the parameter.

Loquacity commented on _partials/_caggs-one-step-policy.mdx:
```diff
@@ -0,0 +1,22 @@
+In TimescaleDB 2.8 and above, policy management on continuous aggregates is
+simplified. You can add, change, or remove the refresh, compression, and data
+retention policies on a continuous aggregate using a one-step API. For more
+information, see the APIs for [adding policies][add-policies], [altering
+policies][alter-policies], and [removing policies][remove-policies]. Note that
+this feature is experimental.
```
```suggestion
information, see the APIs for [adding policies][add-policies], 
[altering policies][alter-policies], and [removing policies][remove-policies]. 
Note that this feature is experimental.
```

Don't we have an experimental block we can add here?

Loquacity commented on _partials/_caggs-one-step-policy.mdx:
```diff
@@ -0,0 +1,22 @@
+this feature is experimental.
+
+<highlight type="note">
+  The policies changed using this API apply to the continuous aggregate, not to
```
```suggestion
  When you change policies with this API, the changes apply to the continuous aggregate, not to
```

Loquacity commented on api/remove_all_policies.md:
```diff
@@ -0,0 +1,54 @@
+
+## Sample usage
+
+Given a continuous aggregate named `example_continuous_aggregate`, remove all
+policies from it. This includes refresh policies, compression policies, and data
+retention policies. It doesn't include user-defined jobs:
```
```suggestion
Remove all policies from a continuous aggregate named 
`example_continuous_aggregate`. This includes refresh policies, compression 
policies, and data retention policies. It doesn't include user-defined jobs:
```

charislam committed:
Apply suggestions from code review

jfjoly commented on _partials/_caggs-one-step-policy.mdx:
```diff
@@ -0,0 +1,27 @@
+import Experimental from "versionContent/_partials/_experimental.mdx";
+
+<h2>
+  Use a one-step policy definition to set a {props.policyType} policy on a
```
Where will this be shown?
Until we get more feedback, we are not trying to move users to the new functions vs. the current process.

charislam replied:
I put it on all the how-to guides for setting refresh, compression, and retention policies. If you don't want to advertise the new functions too broadly, I can take it off the how-to guides for now and just add the API references.

Loquacity commented:
@jfjoly Just waiting on a response from you so we can merge this. Please let us know if you need any changes.

jfjoly commented:
@charislam and I discussed that we will just add the API references. I'm not sure the PR has been updated.

charislam committed:
Remove mentions of one-step policies from how-to guides

Refrain from widely advertising one-step policies until they are more
stable. Keep them in the API references but remove mentions of them from
the how-to guides, so most readers will still do things the old way.
